src/spartan/snark.rs

```diff
@@ -121,6 +141,14 @@ impl<G: Group, EE: EvaluationEngineTrait<G>> RelaxedR1CSSNARKTrait<G> for Relaxe
     let mut cs: SatisfyingAssignment<G> = SatisfyingAssignment::new();
     let _ = circuit.synthesize(&mut cs);
 
+    // Padding variables
+    let num_vars = cs.aux_slice().len();
+
+    (num_vars..num_vars.next_power_of_two()).for_each(|i| {
+      cs.alloc(|| format!("padding_var_{i}"), || Ok(G::Scalar::ZERO))
+        .unwrap();
+    });
+
```

Comment:
Why does the prover only pad variables and not constraints like in the setup function?

Reply:
The variable that this function receives representing the circuit, `circuit`, is produced by `setup` and therefore had its variables and constraints padded elsewhere. The `cs` here is only a `SatisfyingAssignment` (basically the vector (1 || witness || instance)), and so we only need to pad it to the right (one-dimensional) length.